app/scripts/services/Room.js

```diff
@@ -2,6 +2,13 @@
   function Room($firebaseArray) {
     var roomsRef = firebase.database().ref().child("rooms");
     var rooms = $firebaseArray(roomsRef);
+   
+    //My thoughts here were to first create a newRoom variable and set it to null since nothing has been created yet. Then within the function use the $add method to add the new room to the rooms array. I'm currently stuck on how to connect newRoom variable to the newRoom method in ModalCtrl.js
+    var newRoom = null;
+    var createNewRoom = function(){
+      rooms.$add(newRoom);
+    }
```

Comment:
This is exactly, right. To make this a public function for your service, you'll want to add this on line 15 as part of your return:

```javascript
      all: rooms,
      create: function(newRoom) {
        rooms.$add(newRoom); 
      }
```